src/native/libs/System.Native/pal_io.c

```diff
@@ -22,6 +22,7 @@
 #include <sys/file.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
+#include <sys/sysmacros.h>
```

Comment:
This is causing a failure in iOS, iOSSimulator, MacCatalyst, tvOS, OSX and in FreeBSD:
```
 [ 10%] Building C object System.Native/CMakeFiles/System.Native-Static.dir/pal_io.c.o
  /Users/runner/work/1/s/src/native/libs/System.Native/pal_io.c:25:10: fatal error: 'sys/sysmacros.h' file not found
  #include <sys/sysmacros.h>
           ^~~~~~~~~~~~~~~~~
```

I'll try to solve it with:
```c
#if !defined(TARGET_OSX) && !defined(TARGET_BSD)
#include <sys/sysmacros.h>
#endif
```

@steveisok does this look correct to you? Are there any `TARGET_*` checks for iOS, iOSSimulator, MacCatalyst, tvOS, or are they covered by the OSX one?

Reply:
Instead of hard-coding OSX and BSD in if defs, you may want to do a "HAVE" check. Look at some of the other if defs in this file for examples on how to do it. Here's the first one:

https://github.com/dotnet/runtime/blob/17c2cb22a5650d4e7b04f29558ec433b61b45f62/src/native/libs/System.Native/pal_io.c#L35

https://github.com/dotnet/runtime/blob/5978738b58fbf510891f75204d600aa812765a4b/src/native/libs/configure.cmake#L1019-L1037

Reply:
Saw your comment after pushing a new commit with the ifdefs. I'll adjust that code to match your suggestion.

Reply:
I would use

```
#if !defined(__APPLE__)
#include <sys/sysmacros.h>
#endif
```

Instead of

```
#if !defined(TARGET_OSX) && !defined(TARGET_FREEBSD)
#include <sys/sysmacros.h>
#endif
```

That way you'll cover all apple configurations.

Reply:
I'll try the following:

configure.cmake:
```cmake
check_c_source_compiles(
    "
    #include <sys/file.h>

    int main(void)
    {
        makedev(1, 2);
    }
    "
    HAVE_MAKEDEV_SYSFILE)

check_c_source_compiles(
    "
    #include <sys/macros.h>

    int main(void)
    {
        makedev(1, 2);
    }
    "
    HAVE_MAKEDEV_SYSMACROS)

if (NOT HAVE_MAKEDEV_SYSFILE AND NOT HAVE_MAKEDEV_SYSMACROS)
  message(FATAL_ERROR "Cannot find the makedev function on this platform.")
endif()
```

pal_io.c:
```c
#if HAVE_MAKEDEV_SYSMACROS
#include <sys/sysmacros.h>
#endif
```

Reply:
I didn't need check_c_source_compiles. I managed to made it work with check_symbol_exists (I had a typo originally, preventing me from building).